### PR TITLE
[receiver/httpcheck] fail fast on endpoint missing scheme

### LINKED ISSUE #23020: [receiver/httpcheck] receiver should fail fast if endpoint with no scheme is provided
### Component(s)

receiver/httpcheck

### Collector version

v0.78.0

### Is your feature request related to a problem? Please describe.

If I configure the HTTP Check receiver with an endpoint URL without a scheme, the receiver does not fail, but instead produces the `httpcheck.error` metric on each scrape.

```yaml
exporters:
  logging:
    verbosity: detailed

receivers:
  httpcheck:
    endpoint: opentelemetry.io

service:
  pipelines:
    metrics:
      exporters:
      - logging
      receivers:
      - httpcheck
```

Expected output:

```console
$ otelcontribcol --config config.yaml
Error: invalid configuration: receivers::httpcheck: "endpoint" must be in the form of <scheme>://<hostname>:<port>: parse "opentelemetry.io": invalid URI for request
2023/06/01 23:48:58 collector server run finished with error: invalid configuration: receivers::httpcheck: "endpoint" must be in the form of <scheme>://<hostname>:<port>: parse "opentelemetry.io": invalid URI for request
```

Actual result:

```console
$ otelcontribcol --config config.yaml
2023-06-01T23:21:00.657+0200    info    service/telemetry.go:104        Setting up own telemetry...
2023-06-01T23:21:00.657+0200    info    service/telemetry.go:127        Serving Prometheus metrics      {"address": ":8888", "level": "Basic"}
2023-06-01T23:21:00.657+0200    info    exporter@v0.78.2/exporter.go:275        Development component. May change in the future.        {"kind": "exporter", "data_type": "metrics", "name": "logging"}
2023-06-01T23:21:00.658+0200    info    receiver@v0.78.2/receiver.go:296        Development component. May change in the future.        {"kind": "receiver", "name": "httpcheck", "data_type": "metrics"}
2023-06-01T23:21:00.679+0200    info    service/service.go:131  Starting otelcontribcol...      {"Version": "0.78.0-dev", "NumCPU": 16}
2023-06-01T23:21:00.679+0200    info    extensions/extensions.go:30     Starting extensions...
2023-06-01T23:21:00.679+0200    info    service/service.go:148  Everything is ready. Begin running and processing data.
2023-06-01T23:21:03.680+0200    info    MetricsExporter {"kind": "exporter", "data_type": "metrics", "name": "logging", "resource metrics": 1, "metrics": 2, "data points": 6}
2023-06-01T23:21:03.680+0200    info    ResourceMetrics #0
Resource SchemaURL: 
ScopeMetrics #0
ScopeMetrics SchemaURL: 
InstrumentationScope otelcol/httpcheckreceiver 0.78.0-dev
Metric #0
Descriptor:
     -> Name: httpcheck.error
     -> Description: Records errors occurring during HTTP check.
     -> Unit: {error}
     -> DataType: Sum
     -> IsMonotonic: false
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
Data point attributes:
     -> http.url: Str()
     -> error.message: Str(Get "": unsupported protocol scheme "")
StartTimestamp: 2023-06-01 21:21:00.658226525 +0000 UTC
Timestamp: 2023-06-01 21:21:03.6801654 +0000 UTC
Value: 1
Metric #1
Descriptor:
     -> Name: httpcheck.status
     -> Description: 1 if the check resulted in status_code matching the status_class, otherwise 0.
     -> Unit: 1
     -> DataType: Sum
     -> IsMonotonic: false
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
Data point attributes:
     -> http.url: Str()
     -> http.status_code: Int(0)
     -> http.method: Str(GET)
     -> http.status_class: Str(1xx)
StartTimestamp: 2023-06-01 21:21:00.658226525 +0000 UTC
Timestamp: 2023-06-01 21:21:03.6801654 +0000 UTC
Value: 0
NumberDataPoints #1
Data point attributes:
     -> http.url: Str()
     -> http.status_code: Int(0)
     -> http.method: Str(GET)
     -> http.status_class: Str(2xx)
StartTimestamp: 2023-06-01 21:21:00.658226525 +0000 UTC
Timestamp: 2023-06-01 21:21:03.6801654 +0000 UTC
Value: 0
NumberDataPoints #2
Data point attributes:
     -> http.url: Str()
     -> http.status_code: Int(0)
     -> http.method: Str(GET)
     -> http.status_class: Str(3xx)
StartTimestamp: 2023-06-01 21:21:00.658226525 +0000 UTC
Timestamp: 2023-06-01 21:21:03.6801654 +0000 UTC
Value: 0
NumberDataPoints #3
Data point attributes:
     -> http.url: Str()
     -> http.status_code: Int(0)
     -> http.method: Str(GET)
     -> http.status_class: Str(4xx)
StartTimestamp: 2023-06-01 21:21:00.658226525 +0000 UTC
Timestamp: 2023-06-01 21:21:03.6801654 +0000 UTC
Value: 0
NumberDataPoints #4
Data point attributes:
     -> http.url: Str()
     -> http.status_code: Int(0)
     -> http.method: Str(GET)
     -> http.status_class: Str(5xx)
StartTimestamp: 2023-06-01 21:21:00.658226525 +0000 UTC
Timestamp: 2023-06-01 21:21:03.6801654 +0000 UTC
Value: 0
        {"kind": "exporter", "data_type": "metrics", "name": "logging"}
```

### Describe the solution you'd like

The receiver should fail fast during configuration validation, saying that you need to provide a URL with a scheme.

### Describe alternatives you've considered

_No response_

### Additional context

There's already an error message prepared for this scenario [in code](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.78.0/receiver/httpcheckreceiver/config.go#L20), but it's not being used because the `url.Parse` function used in `Validate()` method does not fail on URLs without scheme. I believe we should use `url.ParseRequestURI` function instead.

### PR DESCRIPTION
**Pull request by andrzej-stencel**
Fixes #23020

### CODE REVIEW

**Comment by astencel-sumo** [comment]
@codeboten I've marked this pull request as a `bug_fix` even though I had created the issue as an `enhancement`. Now I'm wondering if this shouldn't be marked as a `breaking` change? :thinking:

File: receiver/httpcheckreceiver/config_test.go [hunk 1]
```diff
@@ -38,11 +38,22 @@ func TestValidate(t *testing.T) {
...
+			expectedErr: multierr.Combine(
+				fmt.Errorf("%w: %s", errInvalidEndpoint, `parse "www.opentelemetry.io/docs": invalid URI for request`),
+			),
+		},
 		{
 			desc: "valid config",
 			cfg: &Config{
 				HTTPClientSettings: confighttp.HTTPClientSettings{
-					Endpoint: "https://opentelemetry.io",
+					Endpoint: "https://opentelemetry.io:80/docs",
```
Comment (codeboten) [inline receiver/httpcheckreceiver/config_test.go:56]: Is this change necessary? I'm assuming `https://opentelemetry.io` is still a valid configuration

File: .chloggen/httpcheck-parse-scheme.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Use this changelog template to create an entry for release notes.
+# If your change doesn't affect end users, such as a test fix or a tooling change,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
```
Comment (codeboten) [inline .chloggen/httpcheck-parse-scheme.yaml:6]: I think bug_fix is probably fine, as it was not working before w/o the scheme

File: receiver/httpcheckreceiver/config_test.go (hunk 1 above)
Comment (astencel-sumo) [reply receiver/httpcheckreceiver/config_test.go:56]: Yes, it's valid. I considered adding a couple different tests for various URLs - with/without port, with/without path, but figured in the end we don't want to be testing the underlying functionality, so just settled on testing a "full-featured" URL. I can revert this change if you want, not even sure what value it brings :wink:

File: .chloggen/httpcheck-parse-scheme.yaml (hunk 2 above)
Comment (astencel-sumo) [reply .chloggen/httpcheck-parse-scheme.yaml:6]: I'm just a bit worried that for a customer that had their collector configured with a couple different receivers including the HTTP Check receiver, they might have not even noticed that the HTTP Check receiver is not working. After this change, when they upgrade they'll see the collector failing to start and might be baffled, not seeing any breaking changes in the Changelog. What do you think?

Comment (codeboten) [reply .chloggen/httpcheck-parse-scheme.yaml:6]: Either i think is fine, but i'm ok if you want to rename this a breaking change

Comment (astencel-sumo) [reply .chloggen/httpcheck-parse-scheme.yaml]: Changed to `breaking`, I think it's better. This component is in `development` status, so introducing breaking changes in it shouldn't be a big deal.

File: receiver/httpcheckreceiver/config_test.go (hunk 1 above)
Comment (codeboten) [reply receiver/httpcheckreceiver/config_test.go:56]: It might be beneficial to have both one w/ the port & without. I know this is just testing the `ParseRequestURI` library under the hood, but i would feel better knowing there's a regression test in the case that someone replaces `ParseRequestURI` with a parsing library that requires the port in the future :D

**Review by codeboten — COMMENTED** [review]
This PR may need a rebase depending on if it's merged before or after https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/23439

File: receiver/httpcheckreceiver/config_test.go (hunk 1 above)
Comment (astencel-sumo) [reply receiver/httpcheckreceiver/config_test.go:65]: Changed so that we still have the previously existing test for `https://opentelemetry.io` endpoint and the `https://opentelemetry.io:80/docs` endpoint as a separate test.

**Review by codeboten — APPROVED** [review]
